src/conversions.rs

```diff
@@ -508,7 +509,10 @@ impl<C: Clone, T: FromJSValConvertible<Config=C>> FromJSValConvertible for Vec<T
 
             for i in 0..length {
                 let mut val = RootedValue::new(cx, UndefinedValue());
-                assert!(JS_GetElement(cx, obj.handle(), i, val.handle_mut()));
+                if !JS_GetElement(cx, obj.handle(), i, val.handle_mut()) {
+                    // On JS exception or error exit loop
```

Comment:
Well, I meant the WPT tests. I thought I saw the automatic exception reporting stuff somewhere in the SpiderMonkey documentation. May be it doesn't. I'll see if I can test.